### PR TITLE
Tile1a JSON

### PR DESCRIPTION
**Pull request by kemcbride**
Write a function that creates tile1a as JSON, save it to data/tiles1a.json, add test to compare deserialized form against generated form

Probably to generate the other tiles, I'll just one-off modify `tile_1a()` and save its output in correspondingly in some named json for that tile.

### CODE REVIEW

**Comment by banool** [comment]
Fixed merge conflicts I introduced.

File: src/load_map.rs [hunk 1]
```diff
@@ -1,24 +1,168 @@
...
+    let sq03 = Square {
+        north_wall: WallType::Impassable,
+        east_wall: WallType::Impassable,
+        south_wall: WallType::Impassable,
+        west_wall: WallType::Clear,
+        square_type: SquareType::PurpleTeleportPad,
+    };
+    my_squares.push(sq03);
+
+    let sq10 = Square {
```
[the remark this answers is not in the document]
Comment (banool) [reply src/load_map.rs:66]: Oh dw this is like an xy.

File: src/load_map.rs [hunk 2]
```diff
@@ -48,4 +192,10 @@ mod tests {
         let serialized = serde_json::to_string(&st).unwrap();
         info!("{}", serialized);
     }
+  
+    pub fn test_1a_and_json_match() {
+        let code1a = tile_1a();
```
Comment (banool) [inline src/load_map.rs:197]: Oh sorry I did a few things:

- Made tests run as part of the pipeline.
- Moved tests into their own module.

But I cooked it, `tile_1a()` here should be `super::tile_1a()` now. Idk if I can pull your PR and fix it.

Comment (kemcbride) [reply src/load_map.rs:197]: i'llll try to take a look

File: src/load_map.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (kemcbride) [reply src/load_map.rs:66]: yeah, kinda gross naming, to be fair

File: data/tiles/1a.json [hunk 3]
```diff
@@ -0,0 +1 @@
...st_wall":0,"south_wall":1,"west_wall":1,"square_type":14},{"north_wall":1,"east_wall":0,"south_wall":0,"west_wall":0,"square_type":0},{"north_wall":5,"east_wall":0,"south_wall":0,"west_wall":0,"square_type":0},{"north_wall":1,"east_wall":1,"south_wall":1,"west_wall":0,"square_type":2},{"north_wall":1,"east_wall":0,"south_wall":1,"west_wall":3,"square_type":0},{"north_wall":0,"east_wall":0,"south_wall":0,"west_wall":0,"square_type":0},{"north_wall":0,"east_wall":0,"south_wall":0,"west_wall":0,"square_type":0},{"north_wall":1,"east_wall":1,"south_wall":1,"west_wall":0,"square_type":1},{"north_wall":1,"east_wall":0,"south_wall":1,"west_wall":1,"square_type":4},{"north_wall":0,"east_wall":0,"south_wall":0,"west_wall":0,"square_type":0},{"north_wall":0,"east_wall":1,"south_wall":0,"west_wall":0,"square_type":0},{"north_wall":1,"east_wall":4,"south_wall":1,"west_wall":1,"square_type":0},{"north_wall":1,"east_wall":0,"south_wall":1,"west_wall":1,"square_type":3},{"north_wall":0,"east_wall":0,"south_wall":2,"west_wall":0,"square_type":0},{"north_wall":0,"east_wall":1,"south_wall":1,"west_wall":0,"square_type":0},{"north_wall":1,"east_wall":1,"south_wall":1,"west_wall":1,"square_type":16}]}
```
[the remark this answers is not in the document]
Comment (kemcbride) [reply data/tiles/1a.json:1]: maybe??? worth trying

**Pull request merged**